tests/functional/codegen/features/iteration/test_for_range.py

```diff
@@ -31,6 +31,71 @@ def repeat(n: uint256) -> uint256:
     assert_tx_failed(lambda: c.repeat(7))
 
 
+def test_range_bound_two_args(get_contract, assert_tx_failed):
```

Comment:
maybe we should add a test where start is a variable but end is a constant, ex. `range(start, 100, bound=5)`

Reply:
To be honest that is not handled in the code at this moment. I based off the existing documentation that specifies this is not supported. I'd be happy to implement this, but maybe a separate PR is better?

Reply:
Oh no :scream: this gives a bad error message
```
@external
def repeat(n: uint256) -> uint256:
    x: uint256 = 0
    for i in range(n, 6, bound=6):
        x += i + 1
    return x
```
Fails compilation with `AssertionError: Number of arguments mismatched: calldataload []`.
This definitely needs improvement, even if unsupported.

Reply:
I've added support and tests for the error messages :+1: